tasks/symlinks.yml

```diff
@@ -0,0 +1,37 @@
+- name: get all subdirectories
+  find:
+    paths: "{{service_item}}/live"
+    recurse: no
+    file_type: directory
+  register: service_subdirs
+  become: yes
+  become_user: root
+
+- name: get all hosts
+  set_fact:
+    service_hosts: "{{service_subdirs.files | map(attribute='path') | map('regex_replace', service_item + '/live/') | list}}"
+
+- name: "stat a fullchain_privkey file in {{service_item}}"
+  stat:
+    path: "{{service_item}}/live/{{item}}/fullchain_privkey.pem"
+  register: src_files
+  loop: "{{service_hosts}}"
+  become: yes
+  become_user: root
+
+- name: "stat a symlink for each hostname in {{haproxy_certs_directory}}"
+  stat:
+    path: "{{haproxy_certs_directory}}/{{item}}.pem"
+  register: dest_symlinks
+  loop: "{{service_hosts}}"
```

Comment:
Why are you checking for existence? The ansible `file` module should handle creating the link and skipping or change as necessary.